include/conversation.php

```diff
@@ -144,222 +144,106 @@ function localize_item(&$item)
+			foreach ($obj->link as $l) {
+				$atts = $l->attributes();
+				switch ($atts['rel']) {
+					case "alternate": $Blink = $atts['href'];
+					case "photo": $Bphoto = $atts['href'];
+				}
 			}
-		}
-
-		$A = '[url=' . Contact::magicLink($Alink) . ']' . $Aname . '[/url]';
-		$B = '[url=' . Contact::magicLink($Blink) . ']' . $Bname . '[/url]';
-		if ($Bphoto != "") {
-			$Bphoto = '[url=' . Contact::magicLink($Blink) . '][img=80x80]' . $Bphoto . '[/img][/url]';
-		}
 
-		/*
-		 * we can't have a translation string with three positions but no distinguishable text
-		 * So here is the translate string.
-		 */
-		$txt = DI::l10n()->t('%1$s poked %2$s');
-
-		// now translate the verb
-		$poked_t = trim(sprintf($txt, "", ""));
-		$txt = str_replace($poked_t, DI::l10n()->t($verb), $txt);
+			$A = '[url=' . Contact::magicLink($Alink) . ']' . $Aname . '[/url]';
+			$B = '[url=' . Contact::magicLink($Blink) . ']' . $Bname . '[/url]';
+			if ($Bphoto != "") {
+				$Bphoto = '[url=' . Contact::magicLink($Blink) . '][img=80x80]' . $Bphoto . '[/img][/url]';
+			}
 
-		// then do the sprintf on the translation string
+			/*
+			 * we can't have a translation string with three positions but no distinguishable text
+			 * So here is the translate string.
+			 */
+			$txt = DI::l10n()->t('%1$s poked %2$s');
 
-		$item['body'] = sprintf($txt, $A, $B). "\n\n\n" . $Bphoto;
+			// now translate the verb
+			$poked_t = trim(sprintf($txt, "", ""));
+			$txt = str_replace($poked_t, DI::l10n()->t($verb), $txt);
 
-	}
+			// then do the sprintf on the translation string
 
-	if ($activity->match($item['verb'],  Activity::TAG)) {
-		$fields = ['author-id', 'author-link', 'author-name', 'author-network',
-			'verb', 'object-type', 'resource-id', 'body', 'plink'];
-		$obj = Item::selectFirst($fields, ['uri' => $item['parent-uri']]);
-		if (!DBA::isResult($obj)) {
-			return;
-		}
+			$item['body'] = sprintf($txt, $A, $B). "\n\n\n" . $Bphoto;
 
-		$author_arr = ['uid' => 0, 'id' => $item['author-id'],
-			'network' => $item['author-network'], 'url' => $item['author-link']];
-		$author  = '[url=' . Contact::magicLinkByContact($author_arr) . ']' . $item['author-name'] . '[/url]';
-
-		$author_arr = ['uid' => 0, 'id' => $obj['author-id'],
-			'network' => $obj['author-network'], 'url' => $obj['author-link']];
-		$objauthor  = '[url=' . Contact::magicLinkByContact($author_arr) . ']' . $obj['author-name'] . '[/url]';
-
-		switch ($obj['verb']) {
-			case Activity::POST:
-				switch ($obj['object-type']) {
-					case Activity\ObjectType::EVENT:
-						$post_type = DI::l10n()->t('event');
-						break;
-					default:
-						$post_type = DI::l10n()->t('status');
-				}
-				break;
-			default:
-				if ($obj['resource-id']) {
-					$post_type = DI::l10n()->t('photo');
-					$m=[]; preg_match("/\[url=([^]]*)\]/", $obj['body'], $m);
-					$rr['plink'] = $m[1];
-				} else {
-					$post_type = DI::l10n()->t('status');
-				}
-				// Let's break everthing ... ;-)
-				break;
```

Comment:
Again a lot code has been killed. This maybe needs more review.

Reply:
It's been indented once to the right, see added lines below.